app/services/post_alerter.rb

```diff
@@ -620,35 +617,62 @@ def notify_pm_users(post, reply_to_user, notified)
   end
 
   def group_notifying_via_smtp(post)
-    return nil if !SiteSetting.enable_smtp || !SiteSetting.enable_imap || post.post_type != Post.types[:regular]
-    post.topic.allowed_groups.where(smtp_enabled: true, imap_enabled: true).first
+    return nil if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
+    post.topic.allowed_groups.where(smtp_enabled: true).first
   end
 
-  def notify_group_direct_emailers(post)
-    email_addresses = []
+  def email_using_group_smtp_if_configured(post)
     emails_to_skip_send = []
     group = group_notifying_via_smtp(post)
-
     return emails_to_skip_send if group.blank?
 
-    # If the post already has an incoming email, it has been set in the
-    # Email::Receiver or via the GroupSmtpEmail job, and thus it was created
-    # via the IMAP/SMTP flow, so there is no need to notify those involved
-    # in the email chain again.
-    if post.incoming_email.blank?
-      email_addresses = post.topic.incoming_email_addresses(group: group)
-      if email_addresses.any?
-        Jobs.enqueue(:group_smtp_email, group_id: group.id, post_id: post.id, email: email_addresses)
-      end
-    end
+    to_address = nil
+    cc_addresses = []
+
+    # We need to use topic_allowed_users here instead of directly_targeted_users
+    # because we want to make sure the to_address goes to the OP of the topic.
+    topic_allowed_users_by_age = post.topic.topic_allowed_users.includes(:user).order(:created_at).reject do |tau|
+      not_allowed?(tau.user, post)
+    end
+    return emails_to_skip_send if topic_allowed_users_by_age.empty?
+
+    # This should usually be the OP of the topic, unless they are the one
+    # replying by email (they are excluded by not_allowed? then)
+    to_address = topic_allowed_users_by_age.first.user.email
+    cc_addresses = topic_allowed_users_by_age[1..-1].map { |tau| tau.user.email }
```

Comment:
I wonder, cannot you just put all emails in `cc` and leave `to` field empty? I think that is valid.

Also, is there a reason why you prefer `cc` over `bcc`? I think we should keep users email addresses private.

Reply:
I think it makes more sense to send a `to` email to the person that opened it and `cc` everyone else, while it might be valid to just cc everyone. Also I don't think there is any issue showing the emails of everyone else. This feature is intended to be used for group inboxes; everyone cc'd on that topic should know each other and be comfortable with emails showing. We have discussed this internally, Zendesk does the same thing.